File: java/bubblesort.java
public class bubblesort {

    public static void bubble(int arr[]){
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            for(int j = 0 ; j < n - i -1 ; j++){
                if(arr[j] > arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int arr[] = {1,2,3,4,5};
        // for (int i = 0; i < arr.length; i++) {
        //     System.out.println(arr[i]);
        // }
        
    }
}

File: CPP/ALL PRACTICE/ca1/bubble_sort.cpp
#include<iostream>
#include<string.h>
using namespace std;

void arr_disp(int arr[], int n ){
    for (int i = 0 ;i <n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

void bubble(int arr[] , int n ){
    for(int i = 0 ; i<n ; i++ ){
        bool swapped = false;
        for(int  j = 0 ; j < n- i - 1 ; j++){
            if(arr[j] > arr[j+1]){
                swap(arr[j], arr[j+1]);
                swapped = true ;
            }
        } 	
        if(!swapped){
            break;
        } 

        arr_disp(arr,n);

    }

} 


                

int main()
{
    int arr[6] = {41,64,1,3,79,32};		 	

    bubble(arr,6);         
                
    return 0;
}